calen.rb

```diff
@@ -0,0 +1,243 @@
+require '/home/suhaib/Desktop/ems/even'
+$events = {}
+########################## Claender Class ########################## 
+class Calender
+    
+    attr_accessor :date
+
+    def initialize(date)
+        @date = date
+    end
+##########################  Add Event ########################## 
+     
+def self.add_event
+        
+        
+        puts 'Enter the day:'
+        d = gets.chomp.to_i
+        until (1..31).include? d
+            
+        puts 'Enter a valid day: '
+        d = gets.chomp.to_i
+         end
+
+        puts 'Enter the Month:'
+        m = gets.chomp.to_i
+        until (1..12).include? m
+        puts 'Enter a valid Month: '
+        m = gets.chomp.to_i
+        end
+
+        puts 'Enter the Year:'
+        y = gets.chomp.to_i
+        until (1900..2050).include? y
+        puts 'Enter a valid Year: '
+        y = gets.chomp.to_i
+        end
+       
+        puts "Enter a title: "
+        title = gets.chomp
+
+        puts "Enter a Description: "
+        desc = gets.chomp
+
+        date = "#{d}-#{m}-#{y}"
+        
+        obj = Calender.new(date)
```

Comment:
Useless initialization of the object. why can't we directly use date?
Why do we need the calendar object?